src/components/Reports/ReportEdit.tsx

```diff
@@ -144,15 +152,17 @@ export default function ReportEdit(): JSX.Element {
       setBusyState(false);
     }
 
-    if (!saveResult || saveResult.requestSucceeded) {
+    if ((!saveResult || saveResult.requestSucceeded) && reportIdInt) {
```

Comment:
-1 check and in other places.. only because we have a path that initializes it to -1
Maybe create a function that checks if reportIdInt is valid and use that in all the places?